File: angularapp/src/app/services/plan.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Plan } from '../model/plan';

@Injectable({
  providedIn: 'root'
})
export class PlanService {
  deleteAddonById(planId: any) {
    throw new Error('Method not implemented.');
  }

  constructor(private http: HttpClient) { }

  public baseUrl = "http://localhost:8001"


  public getlistplan(): Observable<Plan[]> {

    return this.http.get<Plan[]>(this.baseUrl + "/getlistplan");

  }

  public getPlanById(id: number) {

    if (id != null) {

      return this.http.get<Plan>(this.baseUrl + "/getplanbyid/" + id);
    }
  }

  public deletePlanById(planId: any) {

    return this.http.delete(this.baseUrl + "/deleteplanbyid/" + planId);

  }

  public addPlan(plan:Plan){

    return this.http.post<Plan>(this.baseUrl+"/addplan",plan);
  }

  updatePlan(plan:Plan,id:number){
    return this.http.put<Plan>(this.baseUrl+"/update/plan/"+id ,plan)
  }


  public searchPlans(value: any):any {
    return this.http.get<Plan[]>(this.baseUrl + "/searchplan/"+ value);
  }


}

File: springapp/src/main/java/com/example/springapp/service/PlanService.java
package com.example.springapp.service;

import java.util.List;

import com.example.springapp.model.Plan;

public interface PlanService {
	public boolean addPlan(Plan plan);
    public List<Plan> getPlans();
    public Plan updatePlan(Plan plan);
    public Plan getPlanById(Long id);
	public boolean deletePlanById(Long id);
    public boolean updatePlanById(Long planId, Plan plan);
    public List<Plan> searchByPrice(Double price);
}
